ui/jobs/buff_tracker.ts

```diff
@@ -655,7 +700,10 @@ export class BuffTracker {
         existingBuff.clearCooldown(source);
     }
 
-    buff?.onGain(seconds, source);
+    if (option === 'active' && seconds > 0)
+      buff?.onGain(seconds);
+    else if (option === 'cooldown')
+      buff?.onCooldown(seconds, source);
```

Comment:
If you make the above code:
```typescript
if (!buff)
  buff = this.buffs[name] = new Buff(name, info, list, this.options);
```

Then you don't need `buff?.` here and can just say `buff.`